pkg/runtime/wfengine/workflow.go

```diff
@@ -498,6 +504,7 @@ func (wf *workflowActor) runWorkflow(ctx context.Context, actorID string, remind
 		}
 	}
 
+	// [Question] Is this workflow instance completed at this stage, since we are clearing the inbox ?
```

Comment:
yeah seems everything completed correctly and so clearing the inbox for this state.